src/ast/passes/codegen_llvm.cpp

```diff
@@ -1167,15 +1167,28 @@ void CodegenLLVM::visit(Call &call)
                            "cat",
                            AsyncAction::cat);
   } else if (call.func == "exit") {
-    /*
-     * perf event output has: uint64_t asyncaction_id
-     * The asyncaction_id informs user-space that this is not a printf(), but is
-     * a special asynchronous action. The ID maps to exit().
-     */
-    AllocaInst *perfdata = b_.CreateAllocaBPF(b_.getInt64Ty(), "perfdata");
-    b_.CreateStore(b_.getInt64(asyncactionint(AsyncAction::exit)), perfdata);
-    b_.CreateOutput(ctx_, perfdata, sizeof(uint64_t), &call.loc);
-    b_.CreateLifetimeEnd(perfdata);
+    auto elements = AsyncEvent::Exit().asLLVMType(b_);
+    StructType *exit_struct = b_.GetStructType("exit_t", elements, true);
+    AllocaInst *buf = b_.CreateAllocaBPF(exit_struct, "exit");
+    size_t struct_size = datalayout().getTypeAllocSize(exit_struct);
+
+    // Fill in perf event struct.
```

Comment:
Nit:
```suggestion
    // Fill in exit struct.
```
Or just remove this comment - it's not super useful